hisaac/workbench
# Only check for updates to mise if it's been more than 24 hours since the last check

## What's changed and why?

Checking for updates to mise and its plugins takes a little bit of time and can be annoying in a fast moving workflow.

This PR uses a variable named `MISE_UPDATE_CHECK` to determine the last time mise checked for updates. If it has been more than 24 hours since the last check, then an update is run, and the current time is set to `MISE_UPDATE_CHECK` in the local `.env` file.

## How to test

1. Run `just up`.
2. Verify that `mise` updates are checked for and installed if available.
3. Verify that a `.env` file is created at the root of the project if one didn't already exist, and that an accurate value for `MISE_UPDATE_CHECK` is written to it.
4. Run `just up` again to verify that updates are _not_ checked for.
5. Manually modify the value of `MISE_UPDATE_CHECK` to `0`.
6. Run `just up`.
7. Verify that `mise` updates are checked for an installed if available.
8. Verify that an accurate value for `MISE_UPDATE_CHECK` is written to the `.env` file.
